analyzers/src/SonarAnalyzer.CSharp/Rules/DisposeFromDispose.cs

```diff
@@ -68,44 +68,28 @@ protected override void Initialize(SonarAnalysisContext context)
         /// Starting C# 8, "ref structs" (which cannot implement an interface) can also be disposable.
         /// </summary>
         private static bool IsDisposableClassOrStruct(INamedTypeSymbol type, LanguageVersion languageVersion) =>
-            ImplementsDisposable(type) ||
-            type.IsDisposableRefStruct(languageVersion);
+            ImplementsDisposable(type) || type.IsDisposableRefStruct(languageVersion);
 
         private static bool IsCalledInsideDispose(InvocationExpressionSyntax invocation, SemanticModel semanticModel) =>
-            semanticModel.GetEnclosingSymbol(invocation.SpanStart) is IMethodSymbol enclosingMethodSymbol &&
-            IsMethodMatchingDisposeMethodName(enclosingMethodSymbol);
-
-        private Location GetLocation(InvocationExpressionSyntax invocation) =>
-            // We already did the type check before
-            ((MemberAccessExpressionSyntax)invocation.Expression).Name.GetLocation();
+            semanticModel.GetEnclosingSymbol(invocation.SpanStart) is IMethodSymbol enclosingMethodSymbol
+            && IsMethodMatchingDisposeMethodName(enclosingMethodSymbol);
 
         /// <summary>
         /// Verifies that the invocation is calling the correct Dispose() method on an disposable object.
         /// </summary>
         /// <remarks>
         /// Disposable ref structs do not implement the IDisposable interface and are supported starting C# 8.
         /// </remarks>
-        private static bool IsDisposeMethodCalled(InvocationExpressionSyntax invocation, SemanticModel semanticModel, LanguageVersion languageVersion)
-        {
-            if (!(semanticModel.GetSymbolInfo(invocation).Symbol is IMethodSymbol methodSymbol) ||
-                !methodSymbol.IsDisposeMethod())
-            {
-                return false;
-            }
-
-            var disposeMethodSignature = semanticModel.Compilation.GetTypeMethod(SpecialType.System_IDisposable, "Dispose");
-            if (disposeMethodSignature == null)
-            {
-                return false;
-            }
-
-            return methodSymbol.Equals(methodSymbol.ContainingType.FindImplementationForInterfaceMember(disposeMethodSignature)) ||
-                methodSymbol.ContainingType.IsDisposableRefStruct(languageVersion);
-        }
+        private static bool IsDisposeMethodCalled(InvocationExpressionSyntax invocation, SemanticModel semanticModel, LanguageVersion languageVersion) =>
+            semanticModel.GetSymbolInfo(invocation).Symbol is IMethodSymbol methodSymbol
+                && methodSymbol.IsDisposeMethod()
+                && semanticModel.Compilation.GetTypeMethod(SpecialType.System_IDisposable, DisposeMethodName) is { } disposeMethodSignature
+                && (methodSymbol.Equals(methodSymbol.ContainingType.FindImplementationForInterfaceMember(disposeMethodSignature))
+                    || methodSymbol.ContainingType.IsDisposableRefStruct(languageVersion));
```

Comment:
The indent should go -4 spaces here
```suggestion
            && methodSymbol.IsDisposeMethod()
            && semanticModel.Compilation.GetTypeMethod(SpecialType.System_IDisposable, DisposeMethodName) is { } disposeMethodSignature
            && (methodSymbol.Equals(methodSymbol.ContainingType.FindImplementationForInterfaceMember(disposeMethodSignature))
                || methodSymbol.ContainingType.IsDisposableRefStruct(languageVersion));
```